src/domain/balances/entities/balance.entity.ts

```diff
@@ -1,9 +1,18 @@
 import { BalanceToken } from '@/domain/balances/entities/balance.token.entity';
 
-export interface Balance {
-  tokenAddress: string | null;
-  token: BalanceToken | null;
+interface NativeBalance {
+  tokenAddress: null;
+  token: null;
   balance: string;
+}
+
+interface Erc20Balance {
+  tokenAddress: string;
+  token: BalanceToken;
+  balance: string;
+}
+
+export type Balance = (NativeBalance | Erc20Balance) & {
   fiatBalance: string | null;
   fiatConversion: string | null;
-}
+};
```

Comment:
The split is required to maintain type specificity in `BalancesService['_mapBalance']`.